src/zimwriterfs/zimcreatorfs.h

```diff
@@ -29,8 +29,9 @@
 class IHandler
 {
  public:
-  virtual void handleArticle(std::shared_ptr<zim::writer::Article> article) = 0;
-  virtual std::shared_ptr<zim::writer::Article> getMetaArticle() = 0;
+  virtual void handleItem(std::shared_ptr<zim::writer::Item> item) = 0;
```

Comment:
Judging from the commit "Adapt mimetypecounter." (and assuming that it didn't break compilation) the method `IHandler::handleItem()` isn't called (and neither was the old method `IHandler::handleArticle()` called). UPDATE: The answer to this conjecture was found in the previous ("Correctly add file item.") commit (see the comment therein).

Other than that, don't you think that passing the item by a const reference would be more appropriate? Do you envision a handler that will need to store an item passed via `handleItem()` until `getData()` is called.

Reply:
> Other than that, don't you think that passing the item by a const reference would be more appropriate? Do you envision a handler that will need to store an item passed via handleItem() until getData() is called.

It was a time where the ArticleHandler was also use to index the article. We've moved the indexing part in libzim itself and so it left only the mimetype counter.
If we have only one handler, we can probably drop this IHandler and use a more static approach.

I will move code changes in the correct commits but I will not change this right now.
I would like to keep this PR on the adaptation to new api only. Improvement of the code will come in another PRs even if I totally agree with your comment.